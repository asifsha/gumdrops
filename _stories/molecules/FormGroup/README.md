The `<FormGroup>` component is used to wrap your form's inputs.

For a simple text input, it can consist of `<FormGroupLabel>`, `<TextInput>`, and `<FormGroupTextHelp>` components.

However, this depends on what type of input you are working on. Context is optional, and you probably don't need it unless you want to display a hint, warning, or error.

*TextInput example*:

```
<FormGroup context="danger">
    <FormGroupLabel .../>
    <TextInput .../>
    <FormGroupTextHelp .../>
</FormGroup>
```

**Select example**:
```
<FormGroup>
    <FormGroupLabel .../>
    <Select .../>
</FormGroup>
```
