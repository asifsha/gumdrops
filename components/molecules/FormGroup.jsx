import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const FormGroup = ({ isInline, isDisabled, context, className, children, ...otherProps }) => {
    const baseClass = 'gds-form-group';

    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${context}`]: context,
        [`${baseClass}--disabled`]: isDisabled,
        [`${baseClass}--inline`]: isInline
    });

    return (
        <div className={rootClass} {...otherProps}>
            {children}
        </div>
    );
};

FormGroup.displayName = 'FormGroup';

FormGroup.propTypes = {
    isInline: PropTypes.bool,
    isDisabled: PropTypes.bool,
    /** danger, warning, success */
    context: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.node.isRequired
};

export default FormGroup;
