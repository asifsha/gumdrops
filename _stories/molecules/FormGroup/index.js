import React from 'react';
import { boolean, text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';
import readme from './README.md';

import FormGroupLabel from '../../../components/atoms/FormGroupLabel';
import FormGroupTextHelp from '../../../components/atoms/FormGroupTextHelp';
import TextInput from '../../../components/atoms/TextInput';
import Select from '../../../components/atoms/Select';
import FormGroup from '../../../components/molecules/FormGroup';

const contextOptions = {
    danger: 'danger',
    warning: 'warning',
    success: 'success',
    'No Value': ''
};

const selectOptions = [
    {
        name: 'bananas',
        value: 1
    },
    {
        name: 'apples',
        value: 2
    },
    {
        name: 'strawberries',
        value: 3
    },
    {
        name: 'limes',
        value: 4
    }
];

const component = () => (
    <FormGroup
        isInline={boolean('isInline', false)}
        isDisabled={boolean('isDisabled', false)}
        context={optionalSelect('Context', contextOptions, 'success')}
        className={text('Classname', '')}
        style={object('Style', {})}>
        <FormGroupLabel text={text('Input Label', 'Cool Label')} />
        <TextInput name="username" placeholder="Bruce Lee" />
        <FormGroupTextHelp text="You got a cool username right there" />

        <FormGroupLabel text={text('Select label', 'My select label')} />
        <Select options={selectOptions} />
    </FormGroup>
);

export default [readme, component];
