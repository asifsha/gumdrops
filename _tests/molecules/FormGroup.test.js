/* globals mount */
import React from 'react';
import FormGroup from '../../components/molecules/FormGroup';

const defaultProps = {
    isInline: false,
    isDisabled: false,
    context: 'warning',
    className: 'custom-class'
};

describe('Expect <FormGroup>', () => {
    it('to render', () => {
        const wrapper = mount(
            <FormGroup {...defaultProps}>
                <input type="text" />
            </FormGroup>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('to render disabled', () => {
        const props = {
            ...defaultProps,
            isDisabled: true
        };
        const wrapper = mount(
            <FormGroup {...props}>
                <input type="text" />
            </FormGroup>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('to render inline', () => {
        const props = {
            ...defaultProps,
            isInline: true
        };
        const wrapper = mount(
            <FormGroup {...props}>
                <input type="text" />
            </FormGroup>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
