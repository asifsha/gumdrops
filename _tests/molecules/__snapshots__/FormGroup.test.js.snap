// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <FormGroup> to render 1`] = `
<FormGroup
  className="custom-class"
  context="warning"
  isDisabled={false}
  isInline={false}
>
  <div
    className="gds-form-group custom-class gds-form-group--warning"
  >
    <input
      type="text"
    />
  </div>
</FormGroup>
`;

exports[`Expect <FormGroup> to render disabled 1`] = `
<FormGroup
  className="custom-class"
  context="warning"
  isDisabled={true}
  isInline={false}
>
  <div
    className="gds-form-group custom-class gds-form-group--warning gds-form-group--disabled"
  >
    <input
      type="text"
    />
  </div>
</FormGroup>
`;

exports[`Expect <FormGroup> to render inline 1`] = `
<FormGroup
  className="custom-class"
  context="warning"
  isDisabled={false}
  isInline={true}
>
  <div
    className="gds-form-group custom-class gds-form-group--warning gds-form-group--inline"
  >
    <input
      type="text"
    />
  </div>
</FormGroup>
`;
